### PR TITLE
[Refactor]migrate some task dao functions from ProcessServiceImpl to dao modules.

### LINKED ISSUE #12469: [Refactor] Migrate all taskInstance-related interface functions from ProcessServiceImpl
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

+ This is a part of #12403 

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by DarkAssassinator**
## Purpose of the pull request

+ This PR will close https://github.com/apache/dolphinscheduler/issues/12469
+ This is a part of #12403 

## Brief change log

+ Move some taskInstance dao function to DAO module
+ Move some taskDefinition dao functions to DAO moudle
+ Move some taskDefinitionLog dao functions to DAO moudle
+ Move some ProcessInstanceMap dao function to DAO moudle
## Verify this pull request

This pull request is already covered by existing tests

### CODE REVIEW

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/repository/impl/ProcessInstanceMapDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @Override
+    public int createWorkProcessInstanceMap(ProcessInstanceMap processInstanceMap) {
+        int count = 0;
+        if (processInstanceMap != null) {
+            return processInstanceMapMapper.insert(processInstanceMap);
+        }
+        return count;
+    }
```
Comment (caishunfeng) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/repository/impl/ProcessInstanceMapDaoImpl.java]:
```suggestion
    public int createWorkProcessInstanceMap(@NonNull ProcessInstanceMap processInstanceMap) {
        return processInstanceMapMapper.insert(processInstanceMap);
    }
```

**Review by caishunfeng — DISMISSED** [review]
Nice work :+1:

**Comment by caishunfeng** [comment]
rerun the fail CI

**Comment by caishunfeng** [comment]
Hi @DarkAssassinator Please check the fail CI.

**Comment by DarkAssassinator** [comment]
> Hi @DarkAssassinator Please check the fail CI.

ohh, I'm to blame carelessness, after we commit the review suggest, just commit it online, forget add `lombok.NonNull` import, i will fix it tonight. Thx

**Comment by DarkAssassinator** [comment]
> Hi @DarkAssassinator Please check the fail CI.

done

**Review by caishunfeng — APPROVED** [review]

**Pull request merged**